src/platform_impl/macos/window_delegate.rs

```diff
@@ -289,12 +297,8 @@ extern "C" fn window_did_change_screen(this: &Object, _: Sel, _: id) {
 extern "C" fn window_did_change_backing_properties(this: &Object, _: Sel, _: id) {
     trace!("Triggered `windowDidChangeBackingProperties:`");
     with_state(this, |state| {
-        let dpi_factor = unsafe { NSWindow::backingScaleFactor(*state.ns_window) } as f64;
-        if state.previous_dpi_factor != dpi_factor {
-            state.previous_dpi_factor = dpi_factor;
-            state.emit_event(WindowEvent::HiDpiFactorChanged(dpi_factor));
-            state.emit_resize_event();
-        }
+        state.emit_static_hidpi_factor_changed_event();
```

Comment:
This needs to keep track of the previous backing scale factor and not emit the event if the scale factor has not changed.

Reply:
I added tracking into `pub fn emit_static_hidpi_factor_changed_event()`